src/praisonai-deploy/praisonai_deploy/compose.py

```
@@ -0,0 +1,156 @@
+        env["AGENTS_FILE"] = str((project_dir / Path(agents_file).name).resolve())
+
+        cmd = [
+            "docker", "compose",
+            "-f", str(compose_file),
+            "--project-directory", str(project_dir),
+            "up",
+        ]
+        if detach:
+            cmd.append("-d")
+
+        result = subprocess.run(cmd, capture_output=True, text=True, env=env)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)